### PR TITLE
Add blockquote functionality

### PR DESCRIPTION
**Pull request by LeMurphant**
Fix https://github.com/StampyAI/GDocsRelatedThings/issues/64

### CODE REVIEW

File: parser/parser.js [hunk 1]
```diff
@@ -239,7 +239,12 @@ export const mergeSameElements = (elements) =>
 
 export const parseParagraph = (documentContext) => (paragraph) => {
   const { elements, ...paragraphContext } = paragraph;
-  const paragraphStyleName = paragraphContext.paragraphStyle?.namedStyleType;
+  const paragraphStyle = paragraphContext.paragraphStyle || {};
+  const paragraphStyleName = paragraphStyle.namedStyleType;
```
Comment (LeMurphant) [inline parser/parser.js:243]: I'm not sure about this part, it was recommended by Claude

File: parser/parser.js [hunk 2]
```diff
@@ -250,7 +255,7 @@ export const parseParagraph = (documentContext) => (paragraph) => {
   let leadingSpace = "";
 
   // First we check if the "paragraph" is a heading, because the markdown for a heading is the first thing we need to output
-  if (paragraphStyleName.indexOf("HEADING_") === 0) {
+  if (paragraphStyleName?.indexOf("HEADING_") === 0) {
```
Comment (jrhender) [inline parser/parser.js:258]: Is separate bug fix or does it relate to the blockquote functionality?

File: parser/parser.js [hunk 3]
```diff
@@ -296,11 +301,15 @@ export const parseParagraph = (documentContext) => (paragraph) => {
       md.join("").replaceAll("\n", "\n" + leadingSpace + "    ")
     );
   } else {
+    // Add quote marker if the paragraph is indented beyond our threshold
+    const isQuote = indentStart >= QUOTE_INDENT_THRESHOLD;
+    const quotePrefix = isQuote ? "> " : "";
```
Comment (jrhender) [inline parser/parser.js:306]: A bit of a nit so no problem if you want to keep it as is, but a consideration could be that this is introducing a nested conditional (the ternary in the `else`). If text classifications are mutually exclusive (i.e. the text can either be bullet, block quote or normal) then having the conditionals in a single level chain can make this clearer, IMO.

To lift the conditional, the `isQuote` logic could be moved next to where `indentStart` and `QUOTE_INDENT_THRESHOLD` are declared (might be good to have them all together anyways) or a small `isQuote(paragraphStyle)` function could be defined.
```
// Add quote marker if the paragraph is indented beyond our threshold
const isQuote = indentStart >= QUOTE_INDENT_THRESHOLD;
```

Then the conditional would just be
```suggestion
  } else if (isQuote) {
    const quotePrefix = "> ";
```

Note an `else` statement with a separate return would still be necessary for normal text.

**Review by jrhender — APPROVED** [review]
Looks good to me

File: parser/parser.js (hunk 2 above)
Comment (LeMurphant) [reply parser/parser.js:258]: In the previous version, the `?` was integrated directly in the declaration of the variable. This seems to be an update (recommended by Claude) to keep the same functionality. I'm surprised that `namedStyleType` is not used for it, but I don't understand what this part does.

File: parser/parser.js (hunk 3 above)
Comment (LeMurphant) [reply parser/parser.js]: I ended up defining quotePrefix everywhere as an empty string, and updating it if needed. LMK what you think of this type of logic

Comment (jrhender) [reply parser/parser.js]: Yea, current logic looks good to me 👍

**Pull request merged**